core/account-request-handler/src/web2_identity/mod.rs

```diff
@@ -0,0 +1,265 @@
+	> RequestHandler<K, A, S> for Web2IdentityVerification<R>
+{
+	type Response = R;
+
+	fn send_request(&self, request_context: &RequestContext<K, A, S>) -> Result<(), Error> {
+		let mut client = self.make_client()?;
+		let query: Vec<(&str, &str)> =
+			client.query.iter().map(|(k, v)| (k.as_str(), v.as_str())).collect();
+		let response: Self::Response = client
+			.client
+			.get_with::<String, R>(client.path, query.as_slice())
+			.map_err(|e| Error::RquestError(format!("{:?}", e)))?;
+		//TODO log level
+		log::warn!("response:{:?}", response);
+		self.handle_response(request_context, response)
+	}
+
+	fn handle_response(
+		&self,
+		request_context: &RequestContext<K, A, S>,
+		response: Self::Response,
+	) -> Result<(), Error> {
+		{
+			// for testing...
+			let key = storage_double_map_key(
+				"IdentityManagement",
+				"ChallengeCodes",
+				&self.verification_request.target,
+				&StorageHasher::Blake2_128Concat,
+				&self.verification_request.identity,
+				&StorageHasher::Blake2_128Concat,
+			);
+			let mut state = itp_sgx_externalities::SgxExternalities::new();
+
+			state.execute_with(|| {
+				log::warn!("storage--key: {:?}", key);
+				let code: Option<u32> = ita_stf::helpers::get_storage_by_key_hash(key);
+				log::warn!("code: {:?}", code);
+			});
+		}
+
+		let request = &self.verification_request;
+		let payload = response
+			.decrypt_ciphertext(request_context.shielding_key.clone())
+			.map_err(|_| Error::OtherError("decrypt payload error".to_string()))?;
+
+		let user_id = response
+			.get_user_id()
+			.ok_or_else(|| Error::OtherError("can not find user_id".to_string()))?;
+
+		match payload.identity.handle {
+			IdentityHandle::String(ref handle) => {
+				let handle = std::str::from_utf8(handle.as_slice())
+					.map_err(|_| Error::OtherError("convert IdentityHandle error".to_string()))?;
+				if !user_id.eq(handle) {
+					return Err(Error::OtherError("user_id is not the same".to_string()))
+				}
+			},
+			_ => return Err(Error::OtherError("IdentityHandle not support".to_string())),
+		}
+
+		if !payload.identity.eq(&request.identity) {
+			return Err(Error::OtherError("identity is not the same".to_string()))
+		}
+
+		let target_hex = hex::encode(request.target.as_slice());
+		if !payload.owner.eq_ignore_ascii_case(target_hex.as_str()) {
+			return Err(Error::OtherError(format!(
+				"owner is not the same as target:{:?}",
+				target_hex
+			)))
+		}
+
+		if !request.challenge_code.eq(&payload.code) {
+			return Err(Error::OtherError("challenge code is not the same".to_string()))
+		}
+
+		let enclave_account_id = request_context
+			.enclave_signer
+			.get_enclave_account()
+			.map_err(|e| Error::OtherError(format!("{:?}", e)))?;
+
+		let trusted_call = TrustedCall::verify_identity_step2(
+			enclave_account_id,
+			request.target.clone(),
+			request.identity.clone(),
+			ValidationData::Web2(request.validation_data.clone()),
+			request.bn,
+		);
+		submit_call(
```

Comment:
Maybe I miss something here but why don't we simply pass the `request_context` and `request` as a whole as `submit_call` parameters? And maybe we don't need the generic type bound either